### DISCUSSION TITLE
install of Alpine-Vaultwarden LXC not working on Proxmox VE 7.4-13

### DISCUSSION THREAD

**Discussion by BadrEddineMoufid** [tteck/Proxmox#2231] [Help Needed] [answered]
i executed the script and choose the defaults but i get this error at the end and container is running but can't get in to shell from anywhere 

Output:

`
Using Default Settings
Using Distribution: alpine
Using alpine Version: 3.18
Using Container Type: 1
Using Root Password: Automatic Login
Using Container ID: 107
Using Hostname: alpine-vaultwarden
Using Disk Size: 0.3GB
Allocated Cores 1
Allocated Ram 256
Using Bridge: vmbr0
Using Static IP Address: dhcp
Using Gateway Address: Default
Disable IPv6: No
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server Address: Host
Using MAC Address: Default
Using VLAN Tag: Default
Enable Root SSH Access: No
Enable Verbose Mode: No
Creating a Alpine-Vaultwarden LXC using the above default settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 107 was successfully created.
 ✓ Started LXC Container
WARNING: updating and opening https://dl-cdn.alpinelinux.org/alpine/v3.18/main: Permission denied
WARNING: updating and opening https://dl-cdn.alpinelinux.org/alpine/v3.18/community: Permission denied
ERROR: unable to select packages:
  bash (no such package):
    required by: world[bash]

[ERROR] in line 544: exit code 1: while executing command pct exec "$CTID" -- ash -c "apk add bash >/dev/null"
`

**Comment by tteck** [comment]
If you click on either of those links, what is the outcome?

**Reply by BadrEddineMoufid** [reply]
I get an FTP page
![Screenshot 2023-12-23 092309](https://github.com/tteck/Proxmox/assets/45129154/817deb20-4390-4ab2-a519-216266c6515a)
![Screenshot 2023-12-23 092255](https://github.com/tteck/Proxmox/assets/45129154/3482f6c3-9f2c-40fb-8d56-d998954f0111)

**Reply by tteck** [reply]
Then, you should have not gotten the Permission denied error. Try to create the LXC again.

**Reply by BadrEddineMoufid** [reply]
I did several times even just now still I get the same error it first just hangs in there before outputting the first warning to the console for like 2 min

**Reply by tteck** [reply]
The script runs without any problems from this end. 

```
 _    __            ____                          __         
| |  / /___ ___  __/ / /__      ______ __________/ /__  ____ 
| | / / __ `/ / / / / __/ | /| / / __ `/ ___/ __  / _ \/ __ \
| |/ / /_/ / /_/ / / /_ | |/ |/ / /_/ / /  / /_/ /  __/ / / /
|___/\__,_/\__,_/_/\__/ |__/|__/\__,_/_/   \__,_/\___/_/ /_/ 
 Alpine                                                 

Using Advanced Settings
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 101
Using Hostname: alpine-vaultwarden
Using Disk Size: 0.3
Allocated Cores: 1
Allocated RAM: 256
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Alpine-Vaultwarden LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ Downloaded LXC Template
 ✓ LXC Container 101 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.253
 ✓ Internet Connected
 ✓ DNS Resolved github.com to 140.82.113.3
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Installed Alpine-Vaultwarden
 ✓ Installed Web-Vault
 ✓ Started Alpine-Vaultwarden
 ✓ Customized Container
 ✓ Completed Successfully!

Alpine-Vaultwarden should be reachable by going to the following URL.
         http://192.168.86.253:8000 
```

**Reply by BadrEddineMoufid** [reply]
i got another error this time:

`[ERROR] in line 544: exit code 130: while executing command pct exec "$CTID" -- ash -c "apk add bash >/dev/null"`

**Reply by BadrEddineMoufid** [reply]
could this be a firewall problem blocking this stuff the proxmox server is running behind a sophos firewall

**Reply by BadrEddineMoufid** [reply]
I tried installing nginx proxy manager LXC, but I got this : 

`- Creating LXC Container...unable to create CT 108 - unsupported debian version '12.2'
bash: line 163: exit: A problem occured while trying to create container.: numeric argument required
`
i think it's a compatibility issue between Proxmox 7.4 and Debian 12

**Reply by tteck** [reply]
Did you not receive this warning? "Proxmox VE 7 Detected You are currently using Proxmox VE 7, refrain from creating Debian 12 LXCs"

**Reply by BadrEddineMoufid** [reply]
actually yes but I just went with it
something to do with locals 

![Screenshot 2023-12-23 104001](https://github.com/tteck/Proxmox/assets/45129154/abea889a-5390-4822-9353-3f01d8cee1d9)


Edit I tried with another LXC script just to test same result, so am I out of luck I have to upgrade to Version 8

**Reply by tteck** [reply, marked as answer]
Make sure that PVE 7 is fully updated, then you can execute
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/pve8-upgrade.sh)"
```
to easily upgrade to PVE 8

**Reply by BadrEddineMoufid** [reply]
can't risk it running some production stuff and people are working and using it 24/7 so yeah, thanks for your effort, and sorry for wasting your time (the warning was there all the time dumb me)

**Reply by tteck** [reply]
You can use Advanced settings and select Debian 11, everything _should_ work.

**Reply by BadrEddineMoufid** [reply]
thanks I'm trying right now but can't get to set a static IP I get an invalid CIDR address

**Reply by tteck** [reply]
192.168.86.252/24 notice the /24

**Reply by BadrEddineMoufid** [reply]
thanks

**Reply by BadrEddineMoufid** [reply]
question i tried the nginx proxy manager lxc it run but can't access the web interface
i did a port scan only port open are 80, 3000, 22

EDIT: a restart to the container fixed it
